src/libstore/build.cc

```diff
@@ -4374,7 +4384,7 @@ void Worker::waitForInput()
 
 unsigned int Worker::exitStatus()
 {
-    return timedOut ? 101 : (permanentFailure ? 100 : 1);
+    return checkMismatch ? 103 : (hashMismatch ? 102 : (timedOut ? 101 : (permanentFailure ? 100 : 1)));
```

Comment:
It might be time to make this a proper if chain :)

Reply:
Moving the status codes to exception types is also an option, not sure if there's a reason why it's stored in flags separately instead.

Reply:
Maybe we should make this an OR of bits (e.g. 4 == check mismatch, 8 == hash mismatch etc.).

Reply:
Updated. What's the best place to document this, the `nix-store --realise` section?